src/components/OptionsSelector/BaseOptionsSelector.js

```diff
@@ -61,620 +53,652 @@ const defaultProps = {
-        }
-
-        if (prevState.paginationPage !== this.state.paginationPage) {
-            const newSections = this.sliceSections();
-
-            this.setState({
-                sections: newSections,
-            });
-        }
 
-        if (prevState.focusedIndex !== this.state.focusedIndex) {
-            this.focusedOption = this.state.allOptions[this.state.focusedIndex];
-        }
-
-        if (_.isEqual(this.props.sections, prevProps.sections)) {
-            return;
-        }
-
-        const newSections = this.sliceSections();
-        const newOptions = this.flattenSections();
-
-        if (prevProps.preferredLocale !== this.props.preferredLocale) {
-            this.setState({
-                sections: newSections,
-                allOptions: newOptions,
-            });
-            return;
-        }
-        const newFocusedIndex = this.props.selectedOptions.length;
-        const isNewFocusedIndex = newFocusedIndex !== this.state.focusedIndex;
-        const prevFocusedOption = _.find(newOptions, (option) => this.focusedOption && option.keyForList === this.focusedOption.keyForList);
-        const prevFocusedOptionIndex = prevFocusedOption ? _.findIndex(newOptions, (option) => this.focusedOption && option.keyForList === this.focusedOption.keyForList) : undefined;
-        // eslint-disable-next-line react/no-did-update-set-state
-        this.setState(
-            {
-                sections: newSections,
-                allOptions: newOptions,
-                focusedIndex: prevFocusedOptionIndex || (_.isNumber(this.props.focusedIndex) ? this.props.focusedIndex : newFocusedIndex),
-            },
-            () => {
-                // If we just toggled an option on a multi-selection page or cleared the search input, scroll to top
-                if (this.props.selectedOptions.length !== prevProps.selectedOptions.length || (!!prevState.value && !this.state.value)) {
-                    this.scrollToIndex(0);
-                    return;
-                }
-
-                // Otherwise, scroll to the focused index (as long as it's in range)
-                if (this.state.allOptions.length <= this.state.focusedIndex || !isNewFocusedIndex) {
-                    return;
-                }
-                this.scrollToIndex(this.state.focusedIndex);
-            },
-        );
-    }
-
-    componentWillUnmount() {
-        if (this.focusTimeout) {
-            clearTimeout(this.focusTimeout);
-        }
-
-        this.unSubscribeFromKeyboardShortcut();
-    }
+            const indexOfInitiallyFocusedOption = _.findIndex(allOptions, (option) => option.keyForList === props.initiallyFocusedOptionKey);
+
+            return indexOfInitiallyFocusedOption;
+        },
+        [props.shouldTextInputAppearBelowOptions, props.initiallyFocusedOptionKey, props.selectedOptions.length, props.focusedIndex],
+    );
+
+    const isWebOrDesktop = [CONST.PLATFORM.DESKTOP, CONST.PLATFORM.WEB].includes(getPlatform());
```

Comment:
I know you didn't add this, but it's not in compliance with our established code patterns for platform-specific code.